### PR TITLE
Support for alternative current entity for groups

### PR DESCRIPTION
**Pull request by RomRider**
I have a universal media_player which is setup with a sonos connect and external volume control (external amp).
In this case, it's not possible to benefit from the grouping feature offered by mmp. 

This PR introduces a new `real_player` parameter in the `speaker_group` object. If defined, it overrides the `entity` for everything related to group management.

This will work properly once #418 is fixed. (tested using the forked version on which I applied some fixes see https://github.com/mcguiresean/mini-media-player/pull/1)

Feedback appreciated :) (PR seems ugly though...)

### CODE REVIEW

**Comment by kalkih** [comment]
Great!
Please also feel free to create a PR from `RomRider:checkbox-element` to `kalkih:checkbox-element` as I just merged `mcguiresean:checkbox-element`.

**Comment by kalkih** [comment]
I'm a bit unsure about what we should call the option, `realPlayer` doesn't feel optimal, but I also don't have a better suggestion.

An idea on how I think we can make this feature cleaner and basically avoid all changes to `model.js`.
Since we would now send the `this.realPlayer` (if set) into the `mmp-group-list` component, we just have to make sure the MediaPlayerObject assigned to `this.realPlayer` is based on that entity completely (replace all `this.config.entity` in `model.js`) and we should not need all the conditional checks in `model.js` anymore right?

We could sovle this by making an optional parameter (entityId) for the `MediaPlayerObject` constructor which if sent in the constructor we assign to a member variable.

```js
this.entityId = entityId || this.config.entity


// Now instead of this.config.entity we do:
this.entityId
```

**Comment by RomRider** [comment]
Agree that the name is not optimal but I didn't have a better idea either :)
I thought about `alternative_entity` or `alternative_player` or `group_mgmt_entity`. Let me know if you have any other idea.

Regarding the other comment, I'll have a look at your idea. I tried this way initially but it required a lot of changes (exactly what you are saying: avoid using this.config.entity).

**Comment by RomRider** [comment]
This should do it :) It's way better !

File: README.md [hunk 1]
```diff
@@ -145,6 +145,7 @@ See [Speaker group management](#speaker-group-management) for example usage.
 | expanded | boolean | optional | Make the speaker group list expanded by default.
 | show_group_count | boolean | true | Have the number of grouped speakers displayed (if any) in the card name.
 | icon | string | optional | Override default group button icon *(any mdi icon)*.
+| group_mgmt_entity | string | optional | Override the player entity for the group management (Usefull if you use a universal media_player as your entity but still want to use the grouping feature)
```
Comment (kalkih) [inline README.md:148]: usefull -> useful
```suggestion
| group_mgmt_entity | string | optional | Override the player entity for the group management (Useful if you use a universal media_player as your entity but still want to use the grouping feature)
```

**Comment by kalkih** [comment]
Nice, I like it, just the comment above and it's good to go 👍

**Comment by RomRider** [comment]
I'm just fixing a last issue: the lit-element is not updating if there is a change in the groups of the alternative entity.

**Comment by RomRider** [comment]
Argh... I included the other branch... Will fix that, sorry...

**Comment by RomRider** [comment]
All good now :)

**Comment by kalkih** [comment]
Thanks, great work!

**Comment by RomRider** [comment]
Thanks for the merge and feedback 👍

**Pull request merged**